### PR TITLE
double quoting to prevent globbing and word-splitting

### PR DESCRIPTION
**Pull request by bernhard-da**
hi @scotte thx for the nice script. i just quoted the inputs, perhaps you find this useful...

### CODE REVIEW

**Comment by scotte** [comment]
Awesome, thank you! A couple of minor comments, but all-in-all this makes sense and I'm excited to see someone actually using this! :-)

File: borgsnap [hunk 1]
```diff
@@ -103,15 +104,15 @@ dobackup() {
...
   # Expects borgdir, rsyncdir
 
-  mount=$(zfs get -H -o value mountpoint $1)
+  mount=$(zfs get -H -o value mountpoint "$1")
 
   echo "------ $(date) ------"
   echo "Doing local backup of $1@$2"
   BORG_OPTS="-v --stats --compression lz4 --exclude-if-present .noborg"
-  borg create $BORG_OPTS $borgdir::$2 ${mount}/.zfs/snapshot/$2
+  borg create "$BORG_OPTS" "$borgdir"::"$2" "${mount}"/.zfs/snapshot/"$2"
```
Comment (scotte) [inline borgsnap]: How about "${mount}/.zfs/snapshot/$2" instead?

File: borgsnap [hunk 2]
```diff
@@ -103,15 +104,15 @@ dobackup() {
...
   if [[ -n $rsyncdir ]]; then
     echo "Doing rsync backup of $1@$2"
-    borg create $BORG_OPTS --remote-path=borg1 $rsyncdir::$2 ${mount}/.zfs/snapshot/$2
+    borg create "$BORG_OPTS" --remote-path=borg1 "$rsyncdir"::"$2" "${mount}"/.zfs/snapshot/"$2"
```
Comment (scotte) [inline borgsnap]: As above

File: borgsnap [hunk 3]
```diff
@@ -162,42 +163,42 @@ for i in $FS; do
...
     echo "Initializing borg $borgdir"
-    mkdir -p $borgdir
-    borg init $borgdir
+    mkdir -p "$borgdir"
+    borg init "$borgdir"
     echo "Initializing rsync $rsyncdir"
     host=${REMOTE%:*}
     dir=${REMOTE#*:}
-    ssh $host mkdir -p $dir/$pool/$fs
+    ssh "$host" mkdir -p "$dir"/"$pool"/"$fs"
```
Comment (scotte) [inline borgsnap]: As above

File: borgsnap [hunk 4]
```diff
@@ -64,17 +64,18 @@ usage() {
...
 [[ $# == 1 ]] || usage
 
 [[ -r $1 ]] || die "Unable to open $1"
 
-source $1
+source "$1"
 
-export BORG_PASSPHRASE=$(< $PASS)
+BORG_PASSPHRASE=$(< "$PASS")
+export BORG_PASSPHRASE
```
Comment (scotte) [inline borgsnap]: There's enough other bash-specific constructs elsewhere in this script that I don't see a good reason to separate export and assignment.

**Comment by bernhard-da** [comment]
hi @scotte thx for your comments, i've updated the pr

**Comment by scotte** [comment]
I see the rational behind SC2155 too, so I'll put that change back in, along with a bit of additional clean up I saw. Thanks again for your contributions and improvements for this, I appreciate it greatly!

**Comment by scotte** [comment]
These commits have been reverted because some variables were overquoted, breaking the script - it doesn't look like this PR was tested prior to submission.

**Comment by scotte** [comment]
OK, should be good to go now - reinstated the shellcheck code except for what was breaking the script (shellcheck gets some things wrong) and tested the script...

**Pull request merged**